digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java

```diff
@@ -418,7 +442,43 @@ static boolean isDeterministicClientException(Exception ex)
                     return statusCode >= 400 && statusCode < 500;
             }
         }
-        else if (ex instanceof TDClientException) {
+        return isFailedBeforeSendClientException(ex);
+    }
+
+    static boolean isNeedNotRetryException(Exception ex)
+    {
+        if (ex instanceof TDClientHttpException) {
+            int statusCode = ((TDClientHttpException) ex).getStatusCode();
+            switch (statusCode) {
+                case HttpStatus.TOO_MANY_REQUESTS_429:
+                case HttpStatus.REQUEST_TIMEOUT_408:
+                case HttpStatus.UNAUTHORIZED_401:
+                case HttpStatus.FORBIDDEN_403:
+                    return false;
```

Comment:
I am not sure this condition is correct or not.
If this behavior is different from original, we should clarify the change.